funsor/cnf.py

```diff
@@ -70,9 +75,46 @@ def _is_affine(self):
         return True
 
     def unscaled_sample(self, sampled_vars, sample_inputs):
-        if self.red_op in (ops.logaddexp, nullop) and self.bin_op in (ops.logaddexp, ops.add, nullop):
```

Comment:
The previous handling of case `self.bin_op == ops.add` was just barely correct due to subtleties in `Gaussian`. I've made `Gaussian` a little more transparent, and written out more detailed logic here so we can see what is going on.